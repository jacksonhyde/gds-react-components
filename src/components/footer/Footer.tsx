import React, { HTMLProps } from 'react';
import classNames from 'classnames';
import { Container } from '../layout';

interface FooterListProps extends HTMLProps<HTMLOListElement> {
  columns?: boolean;
}

const FooterList: React.FC<FooterListProps> = ({ className, columns, ...rest }) => (
  <ul
    className={classNames(
      'govuk-footer__list',
      { 'govuk-footer__list--three-columns': columns },
      className,
    )}
    {...rest}
  />
);

const FooterListItem: React.FC<HTMLProps<HTMLAnchorElement>> = ({ className, ...rest }) => (
  <li className="govuk-footer__list-item">
    <a className={classNames('govuk-footer__list-item-link', className)} {...rest} />
  </li>
);

const FooterCopyright: React.FC<HTMLProps<HTMLParagraphElement>> = ({ className, ...rest }) => (
  <p className={classNames('govuk-footer__copyright', className)} {...rest} />
);

interface FooterProps extends HTMLProps<HTMLDivElement> {
  visuallyHiddenText?: false | string;
}

interface Footer extends React.FC<FooterProps> {
  List: React.FC<FooterListProps>;
  ListItem: React.FC<HTMLProps<HTMLAnchorElement>>;
  Copyright: React.FC<HTMLProps<HTMLParagraphElement>>;
}

const Footer: Footer = ({
  className, children, visuallyHiddenText, ...rest
}) => (
  <footer {...rest}>
    <div className={classNames('govuk-footer', className)}>
      <Container>
        {visuallyHiddenText ? (
          <h2 className="govuk-u-visually-hidden">{visuallyHiddenText}</h2>
        ) : null}
        {children}
      </Container>
    </div>
  </footer>
);

Footer.defaultProps = {
  visuallyHiddenText: 'Support links',
};

Footer.List = FooterList;
Footer.ListItem = FooterListItem;
Footer.Copyright = FooterCopyright;

export default Footer;
