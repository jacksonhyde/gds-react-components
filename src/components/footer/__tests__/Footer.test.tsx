import React from 'react';
import { shallow, mount } from 'enzyme';
import Footer from '..';

describe('Footer', () => {
  it('matches snapshot', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot('Footer');
    component.unmount();
  });
  it('has default visually hidden text', () => {
    const component = mount(<Footer />);
    expect(component.find('.govuk-u-visually-hidden').text()).toBe('Support links');
    component.unmount();
  });

  it('has disabled visually hidden text', () => {
    const component = mount(<Footer visuallyHiddenText={false} />);
    expect(component.find('.govuk-u-visually-hidden').exists()).toBeFalsy();
    component.unmount();
  });

  it('has custom visually hidden text', () => {
    const component = mount(<Footer visuallyHiddenText="Custom" />);
    expect(component.find('.govuk-u-visually-hidden').text()).toBe('Custom');
    component.unmount();
  });

  describe('Footer.List', () => {
    it('matches snapshot', () => {
      const component = shallow(<Footer.List />);
      expect(component).toMatchSnapshot('Footer.List');
      component.unmount();
    });

    it('adds class when columns', () => {
      const component = shallow(<Footer.List columns />);
      expect(component.hasClass('govuk-footer__list--three-columns')).toBeTruthy();
      component.unmount();
    });
  });

  describe('Footer.ListItem', () => {
    it('matches snapshot', () => {
      const component = shallow(<Footer.ListItem />);
      expect(component).toMatchSnapshot('Footer.ListItem');
      component.unmount();
    });
  });

  describe('Footer.Copyright', () => {
    it('matches snapshot', () => {
      const component = shallow(<Footer.Copyright />);
      expect(component).toMatchSnapshot('Footer.Copyright');
      component.unmount();
    });
  });
});
