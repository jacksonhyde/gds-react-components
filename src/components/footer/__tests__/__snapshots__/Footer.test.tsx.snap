// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer.Copyright matches snapshot: Footer.Copyright 1`] = `
<p
  className="govuk-footer__copyright"
/>
`;

exports[`Footer Footer.List matches snapshot: Footer.List 1`] = `
<ul
  className="govuk-footer__list"
/>
`;

exports[`Footer Footer.ListItem matches snapshot: Footer.ListItem 1`] = `
<li
  className="govuk-footer__list-item"
>
  <a
    className="govuk-footer__list-item-link"
  />
</li>
`;

exports[`Footer matches snapshot: Footer 1`] = `
<footer>
  <div
    className="govuk-footer"
  >
    <Container>
      <h2
        className="govuk-u-visually-hidden"
      >
        Support links
      </h2>
    </Container>
  </div>
</footer>
`;
